hedgehog/src/Hedgehog/Internal/Report.hs

```diff
@@ -869,6 +906,93 @@ ppCoverPercentage :: CoverPercentage -> Doc Markup
 ppCoverPercentage =
   WL.text . renderCoverPercentage
 
+ppCoverBar :: CoverPercentage -> CoverPercentage -> Doc Markup
+ppCoverBar (CoverPercentage percentage) (CoverPercentage minimum_) =
+  let
+    barWidth :: Int
+    barWidth =
+      20
+
+    coverageRatio :: Double
+    coverageRatio =
+      percentage / 100.0
+
+    coverageWidth_ :: Int
+    coverageWidth_ =
+      floor $
+        coverageRatio * fromIntegral barWidth
+
+    minimumRatio :: Double
+    minimumRatio =
+      minimum_ / 100.0
+
+    minimumWidth :: Int
+    minimumWidth =
+      floor $
+        minimumRatio * fromIntegral barWidth
+
+    index :: [a] -> Int
+    index xs =
+      floor $
+        ((coverageRatio * fromIntegral barWidth) - fromIntegral coverageWidth_) *
+        fromIntegral (length xs)
+
+    part xs =
+      xs !! index xs
+
+    fillWidth =
+      barWidth - coverageWidth_ - 1
+
+    fillErrorWidth =
+      max 0 (minimumWidth - coverageWidth_ - 1)
+
+    fillSurplusWidth =
+      fillWidth - fillErrorWidth
+
+    bar :: (Char, [Char]) -> Doc Markup
+    bar (full, parts) =
+      WL.hcat [
+        WL.text $ replicate coverageWidth_ full
+      , if fillWidth >= 0 then
+          if index parts == 0 then
+            if fillErrorWidth > 0 then
+              WL.annotate FailedText $ WL.text [part parts]
+            else
+              WL.annotate CoverageFill $ WL.text [part parts]
+          else
+            WL.text [part parts]
+        else
+          ""
+      , WL.annotate FailedText . WL.text $
+          replicate fillErrorWidth (head parts)
+      , WL.annotate CoverageFill . WL.text $
+          replicate fillSurplusWidth (head parts)
+      --
+      -- Uncomment when debugging:
+      --
+      -- , WL.annotate CoverageFill . WL.text $
+      --        " " ++ show barWidth
+      --     ++ " " ++ show coverageWidth_
+      --     ++ " " ++ show minimumWidth
+      --     ++ " " ++ "/"
+      --     ++ " " ++ show fillErrorWidth
+      --     ++ " " ++ "+"
+      --     ++ " " ++ show fillSurplusWidth
+      --     ++ " " ++ "="
+      --     ++ " " ++ show fillWidth
+      ]
+  in
+    bar ('█', ['·', '▏', '▎', '▍', '▌', '▋', '▊', '▉'])
+
+    -- FIXME Maybe this should be configurable?
```

Comment:
Indeed, it might be a good idea to have this configurable, @jacobstanley.

I've noticed that on (older) Windows machines without WSL/WSL2 and/or earlier versions of Windows 10 2004 (build 19041) the current bars won't render correctly (the last one, basically, `'▉'`).

OTOH, this renders just fine:

```hs
--bar ('⣿', ['·', '⡀', '⡄', '⡆', '⡇', '⣇', '⣧', '⣷'])
```